pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/InteractionRunnerTest.java

```diff
@@ -86,18 +94,41 @@ public void should_throw_exception_when_no_target() {
     }
 
     @org.junit.Test
-    public void should_throw_exception_when_state_param_is_not_empy_nor_map() {
-        when(test.getTestClass()).thenReturn(new TestClass(PactProviderWithWrongStateMethod.class));
-        PactProviderWithWrongStateMethod pactDefinition = new PactProviderWithWrongStateMethod();
-        when(test.getTestInstance()).thenReturn(pactDefinition);
+    public void should_execute_states_with_regular_expression_syntax_for_simple_types() {
 
-        final InteractionRunner interactionRunner = new InteractionRunner();
-        interactionRunner.pactsInstance = pactsInstance;
-        interactionRunner.targetInstance = () -> target;
+        when(providerState.getName()).thenReturn("I have 36 cukes in my belly");
+        final List<ProviderState> providerStates = new ArrayList<>();
+        providerStates.add(providerState);
+        when(requestResponseInteraction.getProviderStates()).thenReturn(providerStates);
+
+        InteractionRunner interactionRunner = new InteractionRunner();
+
+        TestClass testClass = new TestClass(PactProviderWithIntegerParameterStateMethod.class);
+        PactProviderWithIntegerParameterStateMethod test = new PactProviderWithIntegerParameterStateMethod();
+        interactionRunner.executeStateChanges(requestResponseInteraction, testClass, test);
+
+        assertThat(test.getNumberOfCakes())
+                .isEqualTo(36);
+
+    }
+
+    @org.junit.Test
```

Comment:
Nope, there is already a Test class there injected of type `org.jboss.arquillian.test.spi.event.suite.Test`